File: api-spring-performance/src/main/java/com/tcc/victor/api_spring_performance/controller/SensorController.java
package com.tcc.victor.api_spring_performance.controller;

import com.tcc.victor.api_spring_performance.model.Sensor;
import com.tcc.victor.api_spring_performance.service.SensorService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("/api/sensors")
public class SensorController {

    @Autowired
    private SensorService sensorService;

    @PostMapping
    public ResponseEntity<Sensor> create(@RequestBody Sensor sensor) {
        return ResponseEntity.status(HttpStatus.CREATED).body(sensorService.saveSensor(sensor));
    }

    @GetMapping
    public List<Sensor> getAll() {
        return sensorService.getAllSensors();
    }

    @DeleteMapping("/{id}")
    @ResponseStatus(HttpStatus.NO_CONTENT)
    public void delete(@PathVariable String id) {
        sensorService.deleteSensor(id);
    }
}

File: api-nest-performance/src/telemetry/sensor.controller.ts
import { Controller, Get, Post, Body, Param, Delete, HttpCode, HttpStatus } from '@nestjs/common';
import { SensorService } from './sensor.service';
import { Sensor } from './sensor.entity';

@Controller('sensors')
export class SensorController {
  constructor(private readonly sensorService: SensorService) {}

  @Post()
  @HttpCode(HttpStatus.CREATED)
  async create(@Body() sensorData: Partial<Sensor>) {
    return this.sensorService.create(sensorData);
  }

  @Get()
  async findAll() {
    return this.sensorService.findAll();
  }

  @Get(':id')
  async findOne(@Param('id') id: string) {
    return this.sensorService.findOne(id);
  }

  @Delete(':id')
  @HttpCode(HttpStatus.NO_CONTENT)
  async remove(@Param('id') id: string) {
    return this.sensorService.remove(id);
  }
}